src/lib/util/server_http_impl.hpp

```diff
@@ -0,0 +1,482 @@
+// license:MIT
```

Comment:
I didn't really create any of the files here – I split a couple of existing files into interface+inplementation, but I left the copyright and license information as-is. It would probably be best to ask the copyright-holders (Ole Christian Eidheim, Miodrag Milanovic) to make that change.